doc/fundamentals/append_overwrite.md

```diff
@@ -0,0 +1,616 @@
+
+- `--target-file-bytes` will fill each of the split-files with data up to the number of bytes you specify, splitting on the nearest record boundary.  Let's say you have a line-delimited file of 50 lines, with each line having about 20 bytes.  If you use the flags `--split lines --target-file-bytes 100`, you'll see the input file split into about 10 files or so, each of which will have 5 or so lines.  Each split-file's size will hover above the target value of 100 bytes, not going below 100 bytes until the last split-file, which  may be less than 100 bytes.
+
+- `--target-file-datums` will attempt to fill each split-file with the number of datums you specify.  Going back to that same line-delimited 50-line file above, if you use `--split lines --target-file-datums 2`, you'll see the file split into 50 split-files, each of which will have 2 lines.
+
+- Specifying both flags, `--target-file-datums` and `--target-file-bytes`, will result each split-file containing just enough data to satisfy whichever constraint is hit first.  Pachyderm will split the file and then fill the first target split-file with line-based records until it hits the record limit. If it passes the target byte number with just one record, it will move on to the next split-file.  If it hits the target datum number after adding another line, it will move on to the next split-file. Using the example above, if the flags supplied to put-file are `--split lines --target-file-datums 2 --target-file-bytes 100`, it will have the same result as `--target-file-datums 2`, since that's the most compact constraint, and file sizes will hover around 40 bytes.
+
+#### What split data looks like in a Pachyderm repository
+
+Going back to our 50-line file example, let's say that file is named "my-data.txt".  We'll create a repo named "line-data" and load my-data.txt into Pachyderm with the following commands:
+
+```
+$ pachctl create-repo line-data
+
+$ pachctl put-file line-data master -f my-data.txt --split line
+```
+After put-file is complete, list the files in the repo.
+```
+$ pachctl list-file line-data master
+COMMIT                           NAME         TYPE COMMITTED          SIZE
+8cce4de3571f46459cbe4d7fe222a466 /my-data.txt dir  About a minute ago 1.071KiB
+```
+Note that the `list-file` command indicates that the line-oriented file we uploaded, "my-data.txt" is actually a directory. That's because the `--split` flag has instructed Pachyderm to split the file up, and it has created a directory with all the chunks in it.  And, as you can see below,  each chunk will be put into a file.  Those are the split-files.  Each split-file will be given a 16-character filename, left-padded with 0.  Each filename will be numbered sequentially in hexadecimal. We modify the command to list the contents of "my-data.txt", and the output reveals the naming structure used:
```

Comment:
Add a sentence:

Each split-file will automatically be given a 16-character filename, left-padded with 0. Split does not allow you to define more sophisticated file names right now, but this is a feature we plan to add in the future.